### ISSUE TITLE
Flaky `test_aio` CircleCI job due to size check

### ISSUE THREAD

**Issue by AndrewKushnir** [angular/angular#37699]
The `test_aio` CircleCI job sometimes fails false-positively with the payload size error (over 20kb diff):
```
FAIL: Commit c95802f26f80de933f8cf62898115bbe938fff94 uncompressed main-es2015 exceeded expected size by 500 bytes or >1% (expected: 453213, actual: 476811).
 ```
Example of the most recent failure in master [can be found here](https://app.circleci.com/pipelines/github/angular/angular/17542/workflows/f2ff23d5-7841-4a09-8f7b-e7c9e0c72e28/jobs/747016/steps). In that particular case, it was a [docs-only commit](https://github.com/angular/angular/commit/c95802f26f80de933f8cf62898115bbe938fff94).

Restarting the CircleCI job usually fixes the problem.

cc @gkalpak

**Comment by gkalpak** [comment]
FYI, I created #37703 to capture more info to help debug this (the next time it happens).

**Comment by IgorMinar** [comment]
another instance: https://github.com/angular/angular/pull/37686#discussion_r444444799

**Comment by AndrewKushnir** [comment]
New instance of this problem: https://app.circleci.com/pipelines/github/angular/angular/17763/workflows/3e8ecec5-dd47-4d4c-9134-2062500fe1be/jobs/751175/steps

**Comment by IgorMinar** [comment]
two jobs within the same workflow affected by the problem:
- https://app.circleci.com/pipelines/github/angular/angular/17852/workflows/d6d9be42-6fc3-4ebb-a574-2166262f98f1/jobs/752616

And as a bonus - within the same workflow we had the following failures that I cannot reproduce locally:
- `yarn bazel test //integration:hello_world__closure_test` https://app.circleci.com/pipelines/github/angular/angular/17852/workflows/d6d9be42-6fc3-4ebb-a574-2166262f98f1/jobs/752602
- `yarn bazel test --config=ivy //packages/core/test/bundling/todo:symbol_test` https://app.circleci.com/pipelines/github/angular/angular/17852/workflows/d6d9be42-6fc3-4ebb-a574-2166262f98f1/jobs/752604

**Comment by AndrewKushnir** [comment]
> And as a bonus - within the same workflow we had the following failures that I cannot reproduce locally:

@IgorMinar FYI these two additional issues were resolved in https://github.com/angular/angular/commit/98c047b76360218e0b910298416eb1c56b56f163 (in master) and https://github.com/angular/angular/commit/78460c18480a479f72e19c95e5ef6bd655684bfe (in patch).

**Comment by IgorMinar** [comment]
@AndrewKushnir yup. I found out later that those two were actual upstream issues that were reliably reproducible. Thanks for the note though.

**Comment by gkalpak** [comment]
By looking at the diffs of the `main` bundles, this seems to have the same root cause as #37688.
See there for more details, but in a nutshell a second copy of RxJS is sometimes included in the bundle (caused by CLI's `DedupeModuleResolvePlugin`).

**Comment by IgorMinar** [comment]
https://app.circleci.com/pipelines/github/angular/angular/18216/workflows/bd3de66e-a326-480f-8bcf-9145c3892848/jobs/758879/steps - new instance, but I see that we were able to find the root cause and it should be fixed once #37898 lands. yay!
